### PR TITLE
Add Prettier & stricter eslint

### LINKED ISSUE #2458: Configure Prettier formatting
Consistent formatting across the app should improve readability and vscode has a nice plugin which can achieve formatting on file save and make our life as devs much easier.

We are already using prettier in `pi-ui` & `pigui` and it would be great if we could maintain the same prettier config in all apps.

ref: https://prettier.io/

### PR DESCRIPTION
**Pull request by amassarwi**
This diff adds the following:
 - Add prettier & stricter eslint config and pre-commit step to format staged files.
 - Config prettier & add prettier ignore file.
 - Prettify all code base.
 - Fix many many linter errors after addng stricter esilnt rules.

Closes #2458

### CODE REVIEW

File: app/actions/ClientActions.js [hunk 1]
```diff
@@ -190,15 +226,24 @@ const getAccountsBalances = (accounts) => (dispatch, getState) => {
...
     return {
       accountNumber: account.getAccountNumber(),
       accountName: account.getAccountName(),
       externalKeys: account.getExternalKeyCount(),
       internalKeys: account.getInternalKeyCount(),
       importedKeys: account.getImportedKeyCount(),
       hidden: !!hiddenAccounts.find(eq(account.getAccountNumber())),
-      HDPath: "m / 44' / " + chainParams.HDCoinType + "' / " + account.getAccountNumber() + "'",
+      HDPath:
+        "m / 44' / " +
```
Comment (amassarwi) [inline app/actions/ClientActions.js:242]: We could use string template here!

File: app/actions/ClientActions.js [hunk 2]
```diff
@@ -391,14 +492,20 @@ export function updateAccount(account) {
         spendable: 0,
         total: 0,
         votingAuthority: 0,
-        HDPath: "m / 44' / " + chainParams.HDCoinType + "' / " + account.accountNumber + "'",
+        HDPath:
+          "m / 44' / " +
```
Comment (amassarwi) [inline app/actions/ClientActions.js:496]: same here!

File: app/actions/TrezorActions.js [hunk 3]
```diff
@@ -422,19 +507,33 @@ export const walletTxToBtcjsTx = (tx, changeIndex, inputTxs) => async (dispatch,
...
     const inputTx = inputTxsMap[rawHashToHex(inp.getPreviousTransactionHash())];
-    if (!inputTx) throw "Cannot sign transaction without knowing source tx " +
-      rawHashToHex(inp.getPreviousTransactionHash());
-
-    const inputTxOut = inputTx.getOutputsList()[inp.getPreviousTransactionIndex()];
-    if (!inputTxOut) throw sprintf("Trying to use unknown outpoint %s:%d as input",
-      rawHashToHex(inp.getPreviousTransactionHash()), inp.getPreviousTransactionIndex());
+    if (!inputTx)
+      throw (
+        "Cannot sign transaction without knowing source tx " +
```
Comment (amassarwi) [inline app/actions/TrezorActions.js:512]: template string

File: app/actions/TrezorActions.js [hunk 4]
```diff
@@ -422,19 +507,33 @@ export const walletTxToBtcjsTx = (tx, changeIndex, inputTxs) => async (dispatch,
...
+      throw sprintf(
+        "Outpoint %s:%d does not have addresses.",
+        rawHashToHex(inp.getPreviousTransactionHash()),
+        inp.getPreviousTransactionIndex()
+      );
 
     const addrValidResp = await wallet.validateAddress(walletService, addr);
-    if (!addrValidResp.getIsValid()) throw "Input has an invalid address " + addr;
+    if (!addrValidResp.getIsValid())
+      throw "Input has an invalid address " + addr;
```
Comment (amassarwi) [inline app/actions/TrezorActions.js:536]: template string

File: app/actions/TrezorActions.js [hunk 5]
```diff
@@ -733,51 +864,87 @@ export const updateFirmware = (path) => async (dispatch, getState) => {
...
-      (features.major_version < wantMajor) ||
-      (features.major_version == wantMajor && features.minor_version < wantMinor);
+      features.major_version < wantMajor ||
+      (features.major_version == wantMajor &&
+        features.minor_version < wantMinor);
     if (features.model == 1 && versionLessThan(1, 8)) {
-      throw new Error("Trezor Model One needs to run on firmware >= 1.8.0. Found " +
-        features.major_version + "." + features.minor_version + "." + features.patch_version);
+      throw new Error(
+        "Trezor Model One needs to run on firmware >= 1.8.0. Found " +
```
Comment (amassarwi) [inline app/actions/TrezorActions.js:908]: template string

File: app/actions/TrezorActions.js [hunk 6]
```diff
@@ -733,51 +864,87 @@ export const updateFirmware = (path) => async (dispatch, getState) => {
...
+          "." +
+          features.minor_version +
+          "." +
+          features.patch_version
+      );
     } else if (features.model == "T" && versionLessThan(2, 1)) {
-      throw new Error("Trezor Model T needs to run on firmware >= 2.1.0. Found "+
-        features.major_version + "." + features.minor_version + "." + features.patch_version);
+      throw new Error(
+        "Trezor Model T needs to run on firmware >= 2.1.0. Found " +
```
Comment (amassarwi) [inline app/actions/TrezorActions.js:917]: template string

File: app/actions/VersionActions.js [hunk 7]
```diff
@@ -9,72 +9,102 @@ export const GETVERSIONSERVICE_ATTEMPT = "GETVERSIONSERVICE_ATTEMPT";
...
+        const walletVersion = getWalletRPCVersionResponse.getVersionString();
+        ipcRenderer.send("grpc-versions-determined", {
+          requiredVersion,
+          walletVersion
+        });
+        if (!walletVersion) {
+          versionErr = "Unable to obtain Dcrwallet API version";
+        } else if (!semverCompatible(requiredVersion, walletVersion)) {
+            versionErr =
+              "API versions not compatible..  Decrediton requires " +
```
Comment (amassarwi) [inline app/actions/VersionActions.js:77]: template strings in this file

**Comment by matheusd** [comment]
While this is a worthwhile thing to do, this PR is pretty much impossible to review. While have a reasonable trust that there isn't any surprises, I can't easily _verify_ that.

I see you've split the change in a few commits already but there's still some noise in the way. So I'd suggest restructuring this PR in the following ways:

- First, either on one commit (or a series if applicable), fix all non-automatic changes that will be needed after changing the linting rules.
- Then add a commit which updates packages, adds the new rules and introduces the command to prettify stuff.
- Finally on the last commit, add the results of the prettify command.

The net result is that I should be able to, **while on the second-to-last commit**, run `yarn prettify` and get the **exact same results** as the ones committed to in your last commit. This offers much greater assurance that you're not committing anything naughty among your 75K changes.

**Comment by amassarwi** [comment]
@matheusd 
Thanks for you comment.
We be able to replicate my changes with this diff without my re-doing all the manual lint fixes, so if you branch from master and copy over `package.json`, `.eslintrc`, `.prettierrc`, `.prettierignore`.
Then you could run `yarn prettify` & `yarn lint-fix`, then you should end up with same changes I have in this diff, except the manual lint fixes.
Is this sufficient in order to test/review it?

**Comment by amassarwi** [comment]
Side note: you could download vs code Prettier extension and the following chunk to your settings json (`ctrl+;` to open settings) and you will get format on save in vs code:
````
"editor.formatOnSave": true,
"prettier.requireConfig": true,
"editor.defaultFormatter": "esbenp.prettier-vscode"
````

Also, make sure to delete all other old formatting related settings in your settings  json except the ones mentioned above, to avoid conflicts with other extensions.

**Comment by amassarwi** [comment]
Closing this for now and creating to separated PRs easier to replicate.

**Pull request closed without merging**